protoc-gen-kotlin/lib/build.gradle.kts

```diff
@@ -42,18 +42,55 @@ kotlin {
                 implementation(kotlin("test"))
                 implementation(kotlin("test-junit"))
                 implementation("junit:junit:4.12")
+                implementation("com.github.tschuchortdev:kotlin-compile-testing:1.2.9")
             }
         }
     }
 }
 
+configurations {
+    create("downloadProtoc") {
+        isTransitive = false
+    }
+}
+
+dependencies {
+    val arch = System.getProperty("os.arch").toLowerCase().let {
+        when {
+            it.contains("amd64") || it.contains("x86_64") -> "x86_64"
+            else -> "x86_32"
+        }
+    }
+    val os = System.getProperty("os.name").toLowerCase().let {
+        when {
+            it.contains("windows") -> "windows"
+            it.contains("mac os x") || it.contains("darwin") || it.contains("osx") -> "osx"
+            else -> "linux"
+        }
+    }
+    "downloadProtoc"(group="com.google.protobuf", name="protoc", version="3.10.1", classifier = "$os-$arch", ext = "exe")
+}
+
 tasks {
     val generateProto by registering(KotlinProtocTask::class) {
         includeDir.set(project.file("src/commonMain/proto"))
         outputDir.set(project.file("src/commonMain/kotlin"))
         kotlinPackage.set("pbandk.gen.pb")
         logLevel.set("debug")
     }
+
+    val generateTestProto by registering(DescriptorProtocTask::class) {
+        includeDir.set(project.file("src/jvmTest/resources/protos"))
+        outputDir.set(temporaryDir)
```

Comment:
You shouldn't use `temporaryDir` for files that are consumed outside of the task. Gradle is allowed to delete the contents of `temporaryDir` as soon as the task completes. I'd recommend using `File(project.buildDir, name)`. I can't remember for sure, but you might need to `mkdir()` the directory inside a `doFirst` block.